headerlist/header_list.go

```diff
@@ -44,3 +44,61 @@ type Node struct {
 func (n *Node) Prev() *Node {
 	return n.prev
 }
+
+// Equals compares two Nodes if they have the same properties.
+// A structure is comparable only if it contains no slice/array.
+// wire.BlockHeader has Solution byte[]
+func (n *Node) Equals(o Node) bool {
+	if n == nil {
+		return false
+	}
+
+	if n.Height != o.Height {
```

Comment:
Not necessary as Equials(h1, h2) covers all these cases.

Reply:
Not exactly. Equals covers the Height inside of the BlockHeader object, there is Height on Block level as well. On theory its possible Height inside and Height outside to be different,

Reply:
Make sense